src/settings/SettingsSecurity.js

```diff
@@ -1,24 +1,118 @@
-import React from 'react';
+import React, { PropTypes } from 'react';
 import M from '../_common/M';
 import InputGroup from '../_common/InputGroup';
+import * as LiveData from '../_data/LiveData';
 
-export default () => (
-	<div className="mobile-form">
-		<p>
-			<M m="An additional password can be used to restrict access to the cashier."/>
-		</p>
-		<InputGroup
-			id="cashierlockpassword1"
-			label="Cashier password"
-			type="password"
-		/>
-		<InputGroup
-			id="cashierlockpassword2"
-			label="Re-enter your password"
-			type="password"
-		/>
-		<button>
-			<M m="Update" />
-		</button>
-	</div>
-);
+export default class SettingsSecurity extends React.Component {
+
+	static propTypes = {
+		settings: PropTypes.object.isRequired,
+		actions: PropTypes.object.isRequired,
+	};
+
+	static handleUpdateError(response) {
+		if (response.code === 'InputValidationFailed') {
+			let errorDetails;
+			for (const k in response.details) {
+				if (response.details.hasOwnProperty(k)) {
+					errorDetails = errorDetails + `\n${k} ${response.details[k]}`;
+				}
+			}
+		}
+	}
+
+	handleChange(event) {
+		const key = event.target.id;		// code duplication, should be refactored into some mixin
+		const val = event.target.value;
+		const obj = {};
+		obj[key] = val;
+		this.setState(obj);
+	}
+
+    clearState() {
+        this.setState({
+            cashierlockpassword1: '',
+            cashierlockpassword2: '',
+        });
+    }
+
+	onClick() {
+        const state = this.state || {};
```

Comment:
You should set state in the constructor. Then you definitely do not need to initialize it.